src/generic_term.cpp

```diff
@@ -0,0 +1,196 @@
+/*********************                                                        */
+/*! \file generic_term.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Yoni Zohar
+** This file is part of the smt-switch project.
+** Copyright (c) 2020 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Class that maintains the expected
+**        Op and children
+**
+**/
+
+#include "generic_term.h"
+
+#include "exceptions.h"
+#include "utils.h"
+
+using namespace std;
+
+namespace smt {
+
+/* GenericTerm */
+
+GenericTerm::GenericTerm(Sort s, Op o, TermVec c, string r)
+    : sort(s), op(o), children(c), repr(r), is_sym(false), is_par(false)
+{
+  ground = compute_ground();
+}
+
+GenericTerm::GenericTerm(Sort s, Op o, TermVec c, string r, bool is_sym)
+    : sort(s),
+      op(o),
+      children(c),
+      repr(r),
+      is_sym(is_sym),
+      is_par(!is_sym),
+      ground(true)
+{
+}
+
+bool GenericTerm::compute_ground()
```

Comment:
Remind me again, why do we need to know if the term is `ground`?

Reply:
Good point! I will be using this in the generic solver, but for now it is not needed. So I removed it and will add when it is really needed (couple of PRs from now).